galsim/catalog.py

```diff
@@ -312,3 +317,334 @@ def itervalues(self):
     def iteritems(self):
         return self.dict.iteritems()
 
+def makeCOSMOSCatalog(file_name, use_real=True, image_dir=None, dir=None, noise_dir=None,
+                      preload=False, deep_sample=False, exclude_fail=True):
```

Comment:
I don't think this new functionality belongs in the catalog module.  It's not really at all connected with the Catalog or Dict classes.  It could go into real.py, since it's much more connected to that, but I think it actually deserves its own new module.  Perhaps scene.py to use the word you original used for this concept in #635.  Or cosmos.py would also be a good name for it.